src/modules/RA/User/components/UserFields.js

```diff
@@ -1,35 +1,78 @@
 import React from 'react';
-
 import {
   TextInput,
   SimpleForm,
   BooleanInput,
   ReferenceArrayInput,
   SelectArrayInput,
+  useTranslate,
 } from 'react-admin';
 
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Typography from '@material-ui/core/Typography';
+
 import JSONInput from '../../../../components/react-admin/JSONInput';
+import FieldGroup from '../../../../components/react-admin/FieldGroup';
 import UserFieldsHelp from './UserFieldsHelp';
 
 import { RES_USERGROUP } from '../../ra-modules';
 
-const UserFields = ({ edit = false, ...props }) => (
-  <SimpleForm {...props}>
-    {edit && <TextInput disabled source="id" />}
-    {edit && <TextInput disabled source="uuid" />}
+const UserFields = ({ edit = false, ...props }) => {
+  const [advanced, setAdvanced] = React.useState(false);
+  const translate = useTranslate();
+
+  const onAdvancedChange = ({ target: { checked } }) => {
+    setAdvanced(checked);
+  };
+
+  return (
+    <SimpleForm {...props}>
+      {edit && <TextInput disabled source="id" />}
+      {edit && <TextInput disabled source="uuid" />}
+
+      <TextInput source="properties.first_name" label="user.form.firstname" />
+      <TextInput source="properties.last_name" label="user.form.lastname" />
+      <TextInput source="email" type="email" />
+      {!edit && <TextInput source="password" type="password" />}
+      <BooleanInput source="is_superuser" />
```

Comment:
Maybe it should have a condition to display/control this boolean? Such as only superUser can access/edit this setting

Reply:
Real question: User creation shouldn't it be accessible to superUser only?

Reply:
User creation should be accessible to user with "user_management" permissions only. That the point of permissions field that list functional permissions that imply available modules. 
Forbidden modules shouldn't be displayed to user. For now, if a user can manage users, he can create super users. May be later we'll make something to protect that.

Reply:
We discuss this IRL !